ROSA STS creation command clean ups (#34246)
openshift/openshift-docs: modules/rosa-sts-creating-cluster.adoc

$ rosa create cluster --interactive
----

> . Enter the following command to check the status of your cluster and retrieve your cluster ID. During cluster creation,

Reviewer: Consider this edit: To check the status of your cluster and retrieve your cluster ID, enter the next command. The 'State' field changes from 'pending' to 'installing' to 'ready'.

the `State` field from the output will transition from `pending` to `installing`, and finally to `ready`:
+
[source,terminal]
----

> $ rosa describe cluster --cluster=${name}

Reviewer 2: Are the curly braces required as part of the input?

Author: The usage of `${name}` was intended to follow the same usage from the `rosa create cluster` command in https://deploy-preview-34246--osdocs.netlify.app/openshift-rosa/latest/rosa_getting_started/rosa-sts-creating-cluster.html#clipboard-0

====
If installation fails or the `State` field does not change to `ready` after 40 minutes, check the installation troubleshooting documentation for more details.
====

> . Set the environment variable for the `cluster_id` from the previous output:

Reviewer: Consider this edit: Set the environment variable for the 'cluster_id'.

Author: My concern with not mentioning the previous output, is that it may be easy to miss this if they don't know where they're looking.

+
[source,terminal]
----

> $ export cluster_id=<cluster_id>

Reviewer 2: Should this be `--cluster_id=<cluster_id>`?

Author: No

----

[IMPORTANT]
====

> You must wait for the `State` output from `rosa describe cluster` to progress to

Reviewer: Consider this edit: Before proceeding, wait for the "State' output from 'rosa describe cluster' to change to ''pending (Waiting for OIDC configuration)".

> "pending (Waiting for OIDC configuration)" before proceeding.

Reviewer 2: We don;t typically use quotes, so I believe this should be with back ticks instead:  `pending (Waiting for OIDC configuration)`

Reviewer 3: Does that message appear in the terminal like that? If so, JQ is right we need to remove the quotes.

Author: ```suggestion
`pending (Waiting for OIDC configuration)` before proceeding.
```

Author: @sfortner-RH I took the exact output from the earlier `rosa describe cluster` example output. I've already addressed this with backticks 👍

====

> Once the deployment is in a `pending` state, proceed with the remaining cluster setup.

Reviewer: Redundant with above.

> Once the deployment is in a `pending` state, proceed with the remaining cluster setup.

Reviewer 4: If `Once the deployment is in a `pending` state, proceed with the remaining cluster setup.` is required here. I prefer keep `pending` consist with above description `pending (Waiting for OIDC configuration)`

Author: Already mentioned

> . Complete setup of the OIDC access-based IAM roles.

Reviewer: Add _the_ before _setup_.

> You can complete these steps once the `State` field for the cluster creation transitions to `pending`.

Reviewer: Already stated. Ok to delete.

> You can complete these steps once the `State` field for the cluster creation transitions to `pending`.

Reviewer 4: `You can complete these steps once the State field for the cluster creation transitions to pending.` seems redundant with above statement

Author: Already mentioned

$ cat credrequests/0000*.yaml > credrequests/${version:0:3}.yaml
$ rm -f credrequests/0000*.yaml
----
+
[NOTE]
====

> This requires the OpenShift CLI (`oc`), version 4.7.9 or greater. You can

Reviewer 2: Right now, cross-references are not supported from modules. So you may want to just give them the download command instead.

> This requires the OpenShift CLI (`oc`), version 4.7.9 or greater. You can

Reviewer: Add _action_ after _This_.

> xref:../rosa_cli/rosa-get-started-cli.adoc#rosa-download-ocp-client_rosa-getting-started-cli[download the latest `oc` version]

Reviewer 2: Right now, cross-references are not supported from modules. So you may want to just give them the download command instead.

Author: Last I knew, they _were_ supported in modules. @vikram-redhat can you confirm?

do 
  policy=$(sed -e 's/05-/06-/' -e 's/role/policy/' <<< ${role})
  role_name=$(grep RoleName ${policy} | awk '{print $2}' | awk -F '"' '{print $2}')
  aws iam create-role --cli-input-json file://${role}
  sed -i.bak '/RoleName/d' ${policy}
  policy_arn=$(aws iam create-policy --cli-input-json file://$policy | grep Arn | awk '{print $2}' | awk -F '"' '{print $2}')
  aws iam attach-role-policy --role-name $role_name --policy-arn $policy_arn
  rm ${policy}
  mv ${policy}.bak ${policy}
  sleep 5 # Prevents AWS Rate limiting
done
----
+
[NOTE]
====

> If relying on a permissions boundary arn, use the following `aws iam create-role` command in the above loop

Reviewer 2: s\ above \ previous

Author: ```suggestion
If relying on a permissions boundary arn, use the following `aws iam create-role` command in the previous loop
```

> If relying on a permissions boundary arn, use the following `aws iam create-role` command in the above loop

Reviewer: ARN all caps?

[source,terminal]
----
$ aws iam create-role \
    --cli-input-json file://${role} \
    --permissions-boundary ${permissions_boundary_arn}
----
====
+

> . Enter the following command to follow the OpenShift installer logs to track the progress of your cluster creation:

Reviewer 2: s \ to track the progress \ that track the progress

"to" is used twice in this sentence and it just seems a bit awkward to my eyes.

Author: ```suggestion
. Enter the following command to follow the OpenShift installer logs that tracks the progress of your cluster creation:
```

> . Enter the following command to follow the OpenShift installer logs to track the progress of your cluster creation:

Reviewer: Consider this edit: To track the progress of your cluster creation, enter this command to watch the OpenShift installer logs.

+
[source,terminal]
----

> $ rosa logs install --cluster=${name} --watch

Reviewer 2: Same question as before about the curly braces
